benchmarks/src/bitset_to_string.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <array>
+#include <benchmark/benchmark.h>
+#include <bit>
+#include <bitset>
+#include <random>
+
+using namespace std;
+
+namespace {
+    const auto random_bits = [] {
+        mt19937_64 rnd{};
+        array<uint64_t, 32> data;
+        for (auto& d : data) {
+            d = rnd();
+        }
+        return data;
+    }();
+
+    template <size_t N, class charT>
+    void BM_bitset_to_string(benchmark::State& state) {
+        for (auto _ : state) {
+            for (auto bits : random_bits) {

```

Comment:
Stylistic nitpick: Technically, this is a modifiable copy, which is slightly risky. `const auto&` iterates in-place which is basically always desirable. (There's no performance cost because iteration is already done through pointers.) This isn't necessary for `auto _ : state` but we conventionally use `[const] auto&` for all other range-fors.